---
created: '2021-03-18T11:30:02.505718Z'
creator: sentry
source: tests/sentry/grouping/test_variants.py
---
app-depth-1:
  hash: "75c06f01cb4cb57f79d076bcdece59e7"
  tree_label: "Get"
  component:
    app-depth-1*
      exception*
        stacktrace*
          frame*
            module*
              "SentryTest2.Controllers.ValuesController"
            filename (module takes precedence)
              "valuescontroller.cs"
            function*
              "Get"
            package (ignored because function takes precedence)
        value (ignored because stacktrace takes precedence)
          "sync exception"
--------------------------------------------------------------------------
app-depth-2:
  hash: "228c649a3aa0901622c0a0e66ab0522c"
  tree_label: "lambda_method | ..."
  component:
    app-depth-2*
      exception*
        stacktrace*
          frame*
            module*
              "(unknown)"
            function*
              "lambda_method"
          frame*
            module*
              "SentryTest2.Controllers.ValuesController"
            filename (module takes precedence)
              "valuescontroller.cs"
            function*
              "Get"
            package (ignored because function takes precedence)
        type*
          "System.Exception"
        value (ignored because stacktrace takes precedence)
          "sync exception"
--------------------------------------------------------------------------
app-depth-3:
  hash: "1a74c747b5cea2b8c2ac2f5753a03116"
  tree_label: "Execute | ..."
  component:
    app-depth-3*
      exception*
        stacktrace*
          frame*
            module*
              "Microsoft.Extensions.Internal.ObjectMethodExecutor"
            function*
              "Execute"
            package (ignored because function takes precedence)
          frame*
            module*
              "(unknown)"
            function*
              "lambda_method"
          frame*
            module*
              "SentryTest2.Controllers.ValuesController"
            filename (module takes precedence)
              "valuescontroller.cs"
            function*
              "Get"
            package (ignored because function takes precedence)
        type*
          "System.Exception"
        value (ignored because stacktrace takes precedence)
          "sync exception"
--------------------------------------------------------------------------
app-depth-4:
  hash: "14a57462092f01a0042382359f958006"
  tree_label: "Execute | ..."
  component:
    app-depth-4*
      exception*
        stacktrace*
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ActionMethodExecutor+SyncObjectResultExecutor"
            function*
              "Execute"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.Extensions.Internal.ObjectMethodExecutor"
            function*
              "Execute"
            package (ignored because function takes precedence)
          frame*
            module*
              "(unknown)"
            function*
              "lambda_method"
          frame*
            module*
              "SentryTest2.Controllers.ValuesController"
            filename (module takes precedence)
              "valuescontroller.cs"
            function*
              "Get"
            package (ignored because function takes precedence)
        type*
          "System.Exception"
        value (ignored because stacktrace takes precedence)
          "sync exception"
--------------------------------------------------------------------------
app-depth-5:
  hash: "8bd7d4553a8f5d58ad65bea951793080"
  tree_label: "InvokeActionMethodAsync | ..."
  component:
    app-depth-5*
      exception*
        stacktrace*
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker"
            function*
              "InvokeActionMethodAsync"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ActionMethodExecutor+SyncObjectResultExecutor"
            function*
              "Execute"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.Extensions.Internal.ObjectMethodExecutor"
            function*
              "Execute"
            package (ignored because function takes precedence)
          frame*
            module*
              "(unknown)"
            function*
              "lambda_method"
          frame*
            module*
              "SentryTest2.Controllers.ValuesController"
            filename (module takes precedence)
              "valuescontroller.cs"
            function*
              "Get"
            package (ignored because function takes precedence)
        type*
          "System.Exception"
        value (ignored because stacktrace takes precedence)
          "sync exception"
--------------------------------------------------------------------------
app-depth-max:
  hash: "4ccd0f1953483581ba360c7518f90332"
  tree_label: "<entire stacktrace>"
  component:
    app-depth-max*
      exception*
        stacktrace*
          frame*
            module*
              "Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware"
            function*
              "Invoke"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.CompilerServices.TaskAwaiter"
            function*
              "HandleNonSuccessAndDebuggerNotification"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.ExceptionServices.ExceptionDispatchInfo"
            function*
              "Throw"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware"
            function*
              "Invoke"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.CompilerServices.TaskAwaiter"
            function*
              "HandleNonSuccessAndDebuggerNotification"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.ExceptionServices.ExceptionDispatchInfo"
            function*
              "Throw"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Routing.EndpointMiddleware"
            function*
              "Invoke"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.CompilerServices.TaskAwaiter"
            function*
              "HandleNonSuccessAndDebuggerNotification"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.ExceptionServices.ExceptionDispatchInfo"
            function*
              "Throw"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker"
            function*
              "InvokeAsync"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.CompilerServices.TaskAwaiter"
            function*
              "HandleNonSuccessAndDebuggerNotification"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.ExceptionServices.ExceptionDispatchInfo"
            function*
              "Throw"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker"
            function*
              "InvokeFilterPipelineAsync"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker"
            function*
              "Next"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker"
            function*
              "Rethrow"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.ExceptionServices.ExceptionDispatchInfo"
            function*
              "Throw"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker"
            function*
              "InvokeNextResourceFilter"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.CompilerServices.TaskAwaiter"
            function*
              "HandleNonSuccessAndDebuggerNotification"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.ExceptionServices.ExceptionDispatchInfo"
            function*
              "Throw"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker"
            function*
              "InvokeInnerFilterAsync"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker"
            function*
              "Next"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker"
            function*
              "Rethrow"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.ExceptionServices.ExceptionDispatchInfo"
            function*
              "Throw"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker"
            function*
              "InvokeNextActionFilterAsync"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.CompilerServices.TaskAwaiter"
            function*
              "HandleNonSuccessAndDebuggerNotification"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.ExceptionServices.ExceptionDispatchInfo"
            function*
              "Throw"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker"
            function*
              "InvokeActionMethodAsync"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ActionMethodExecutor+SyncObjectResultExecutor"
            function*
              "Execute"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.Extensions.Internal.ObjectMethodExecutor"
            function*
              "Execute"
            package (ignored because function takes precedence)
          frame*
            module*
              "(unknown)"
            function*
              "lambda_method"
          frame*
            module*
              "SentryTest2.Controllers.ValuesController"
            filename (module takes precedence)
              "valuescontroller.cs"
            function*
              "Get"
            package (ignored because function takes precedence)
        type*
          "System.Exception"
        value (ignored because stacktrace takes precedence)
          "sync exception"
--------------------------------------------------------------------------
default:
  hash: null
  component:
    default (exception of app-depth-1/app-depth-2/app-depth-3/app-depth-4/app-depth-5/app-depth-max/exception-type/system takes precedence)
      message (exception of app-depth-1/app-depth-2/app-depth-3/app-depth-4/app-depth-5/app-depth-max/exception-type/system takes precedence)
        "An unhandled exception has occurred while executing the request."
--------------------------------------------------------------------------
exception-type:
  hash: "d1546df68cd784043f201b7375878b7a"
  tree_label: "System.Exception"
  component:
    exception-type*
      exception-type*
        type*
          "System.Exception"
--------------------------------------------------------------------------
system:
  hash: "4ccd0f1953483581ba360c7518f90332"
  tree_label: "<entire stacktrace>"
  component:
    system*
      exception*
        stacktrace*
          frame*
            module*
              "Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware"
            function*
              "Invoke"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.CompilerServices.TaskAwaiter"
            function*
              "HandleNonSuccessAndDebuggerNotification"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.ExceptionServices.ExceptionDispatchInfo"
            function*
              "Throw"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware"
            function*
              "Invoke"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.CompilerServices.TaskAwaiter"
            function*
              "HandleNonSuccessAndDebuggerNotification"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.ExceptionServices.ExceptionDispatchInfo"
            function*
              "Throw"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Routing.EndpointMiddleware"
            function*
              "Invoke"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.CompilerServices.TaskAwaiter"
            function*
              "HandleNonSuccessAndDebuggerNotification"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.ExceptionServices.ExceptionDispatchInfo"
            function*
              "Throw"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker"
            function*
              "InvokeAsync"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.CompilerServices.TaskAwaiter"
            function*
              "HandleNonSuccessAndDebuggerNotification"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.ExceptionServices.ExceptionDispatchInfo"
            function*
              "Throw"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker"
            function*
              "InvokeFilterPipelineAsync"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker"
            function*
              "Next"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker"
            function*
              "Rethrow"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.ExceptionServices.ExceptionDispatchInfo"
            function*
              "Throw"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker"
            function*
              "InvokeNextResourceFilter"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.CompilerServices.TaskAwaiter"
            function*
              "HandleNonSuccessAndDebuggerNotification"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.ExceptionServices.ExceptionDispatchInfo"
            function*
              "Throw"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker"
            function*
              "InvokeInnerFilterAsync"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker"
            function*
              "Next"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker"
            function*
              "Rethrow"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.ExceptionServices.ExceptionDispatchInfo"
            function*
              "Throw"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker"
            function*
              "InvokeNextActionFilterAsync"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.CompilerServices.TaskAwaiter"
            function*
              "HandleNonSuccessAndDebuggerNotification"
            package (ignored because function takes precedence)
          frame*
            module*
              "System.Runtime.ExceptionServices.ExceptionDispatchInfo"
            function*
              "Throw"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker"
            function*
              "InvokeActionMethodAsync"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.AspNetCore.Mvc.Internal.ActionMethodExecutor+SyncObjectResultExecutor"
            function*
              "Execute"
            package (ignored because function takes precedence)
          frame*
            module*
              "Microsoft.Extensions.Internal.ObjectMethodExecutor"
            function*
              "Execute"
            package (ignored because function takes precedence)
          frame*
            module*
              "(unknown)"
            function*
              "lambda_method"
          frame*
            module*
              "SentryTest2.Controllers.ValuesController"
            filename (module takes precedence)
              "valuescontroller.cs"
            function*
              "Get"
            package (ignored because function takes precedence)
        type*
          "System.Exception"
        value (ignored because stacktrace takes precedence)
          "sync exception"
